xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/TemplateProviderSheet.xml

```diff
@@ -144,6 +144,46 @@
 
 {{html clean='false'}}
       &lt;/dd&gt;
+      ##
+      ## Description
+      ##
+      &lt;dt&gt;
+        &lt;label #if($isEditing)for="XWiki.TemplateProviderClass_0_description"#end&gt;$escapetool.xml($services.localization.render('xe.templateprovider.description'))&lt;/label&gt;
+      &lt;/dt&gt;
+      &lt;dd&gt;
+        #if($isEditing)
+          {{/html}}
+
+          {{html wiki='true' clean='false'}}$!providerObj.get('description'){{/html}}
```

Comment:
This document is full of this usage and, IMO, requires a revision. I followed the implementation of the other properties. I may take some time to investigate and clean this, if is needed. Thanks!